### PR TITLE
[Remove Spring extender usages] LPS-159168

### PR DESCRIPTION
**Pull request by JanisZhang**
Hi @dantewang @tinatian,

This PR is to remove spring extender usage in commerce-currency-service, thanks!
https://issues.liferay.com/browse/LPS-159168

Previous PR:https://github.com/liferay-core-infra/liferay-portal/pull/1222

### CODE REVIEW

**Comment by JanisZhang** [comment]
Hi @catarineaguiar, Could you review the test results when they finish? Thanks!

File: modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/service/impl/CommerceCurrencyLocalServiceImpl.java [hunk 1]
```diff
@@ -463,6 +441,22 @@ public void updateExchangeRates() throws PortalException {
...
+	protected void activate(BundleContext bundleContext) {
+		_serviceRegistration = bundleContext.registerService(
+			PortalInstanceLifecycleListener.class,
+			new PortalInstanceLifecycleListenerImpl(
+				commerceCurrencyLocalService),
+			null);
+	}
+
+	@Deactivate
+	protected void deactivate() {
```
Comment (dantewang) [inline modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/service/impl/CommerceCurrencyLocalServiceImpl.java:454]: @JanisZhang The BaseImpl has a deactivate() method, so you should add `@Override` and `super.deactivate()`.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by JanisZhang** [comment]
ci:stop

**Comment by JanisZhang** [comment]
@dantewang force-pushed based on the comment. Thanks!

**Comment by JanisZhang** [comment]
@dantewang force-pushed based on the comment. Thanks!

**Comment by JanisZhang** [comment]
ci:test:sf

**Comment by JanisZhang** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
all the PR analysis on CI were aborted according to this [message](https://liferay.slack.com/archives/C19PWQ74J/p1659633528796189) 

when this runs finish, will need to re-run relevant.

**Comment by catarineaguiar** [comment]
@JanisZhang regressions found:

`CommercePricingClassLocalServiceTest`
```
  [exec] com.liferay.commerce.pricing.service.test.CommercePricingClassLocalServiceTest > testAddProductToPricingClass FAILED
     [exec]     com.liferay.commerce.currency.exception.NoSuchCurrencyException: 
     [exec]     [
     [exec]     SCENARIO A CPDefinition is added to a pricing class
     [exec]     GIVEN A product in a catalog
     [exec]      AND A pricing class
     [exec]     WHEN The product is added to the pricing class
     [exec]     THEN It should be returned when querying the pricing class content
     [exec]     ]
     [exec]     [
     [exec]     Message: No CommerceCurrency exists with the key {companyId=20098, code=USD}
     [exec]     ]
     [exec]     	at com.liferay.commerce.currency.service.persistence.impl.CommerceCurrencyPersistenceImpl.findByC_C(CommerceCurrencyPersistenceImpl.java:1754)
     [exec]     	at com.liferay.commerce.currency.service.impl.CommerceCurrencyLocalServiceImpl.getCommerceCurrency(CommerceCurrencyLocalServiceImpl.java:211)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
```
full log [here](https://testray.liferay.com/reports/production/logs/2022-08/test-1-30/test-portal-acceptance-pullrequest(master)/4042/modules-integration-mysql57-jdk8/0/4/jenkins-console.txt.gz?authuser=0)

`CSDiagramPinTableReferenceDefinitionTest`
```
 [exec] com.liferay.commerce.shop.by.diagram.change.tracking.test.CSDiagramPinTableReferenceDefinitionTest > testDiscardCTEntry FAILED
     [exec]     com.liferay.commerce.currency.exception.NoSuchCurrencyException: No CommerceCurrency exists with the key {companyId=20098, code=USD}
     [exec]     	at com.liferay.commerce.currency.service.persistence.impl.CommerceCurrencyPersistenceImpl.findByC_C(CommerceCurrencyPersistenceImpl.java:1754)
     [exec]     	at com.liferay.commerce.currency.service.impl.CommerceCurrencyLocalServiceImpl.getCommerceCurrency(CommerceCurrencyLocalServiceImpl.java:211)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
     [exec]     	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     [exec]     	at java.lang.reflect.Method.invoke(Method.java:498)
     [exec]     	at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:50)
```
full log [here](https://testray.liferay.com/reports/production/logs/2022-08/test-1-30/test-portal-acceptance-pullrequest(master)/4042/modules-integration-mysql57-jdk8/0/7/jenkins-console.txt.gz?authuser=0)

`CommerceOrderImporterTypeTest`
```
[exec] com.liferay.commerce.order.importer.type.test.CommerceOrderImporterTypeTest > testSuccessfulCSVImport FAILED
     [exec]     java.lang.AssertionError: expected null, but was:<[Ljava.lang.String;@25ed7ce5>
     [exec]     	at org.junit.Assert.fail(Assert.java:89)
     [exec]     	at org.junit.Assert.failNotNull(Assert.java:756)
     [exec]     	at org.junit.Assert.assertNull(Assert.java:738)
     [exec]     	at org.junit.Assert.assertNull(Assert.java:748)
     [exec]     	at com.liferay.commerce.order.importer.type.test.CommerceOrderImporterTypeTest.testSuccessfulCSVImport(CommerceOrderImporterTypeTest.java:257)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
```
full log [here](https://testray.liferay.com/reports/production/logs/2022-08/test-1-30/test-portal-acceptance-pullrequest(master)/4042/modules-integration-mysql57-jdk8/0/8/jenkins-console.txt.gz?authuser=0)

`CommerceInventoryMVCCTest`
```
   [exec] com.liferay.commerce.inventory.service.mvcc.test.CommerceInventoryMVCCTest > testReplenishmentItemMVCC FAILED
     [exec]     java.lang.Exception: Unexpected exception, expected<com.liferay.commerce.inventory.exception.MVCCException> but was<com.liferay.commerce.currency.exception.NoSuchCurrencyException>
     [exec]     	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable._processExpectedException(TestExecutorRunnable.java:416)
     [exec]     	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable.access$400(TestExecutorRunnable.java:66)
     [exec]     	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable$3.evaluate(TestExecutorRunnable.java:380)
     [exec]     	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
     [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
```
full log [here](https://testray.liferay.com/reports/production/logs/2022-08/test-1-30/test-portal-acceptance-pullrequest(master)/4042/modules-integration-mysql57-jdk8/0/1/jenkins-console.txt.gz?authuser=0)

**Pull request closed without merging**